extensions/azurecore/src/azureResource/providers/postgresArcServer/postgresServerTreeDataProvider.ts

```diff
@@ -0,0 +1,82 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ExtensionNodeType, TreeItem } from 'azdata';
+import { TreeItemCollapsibleState, ExtensionContext } from 'vscode';
+import * as nls from 'vscode-nls';
+const localize = nls.loadMessageBundle();
+
+import { AzureResourceItemType } from '../../constants';
+import { ApiWrapper } from '../../../apiWrapper';
+import { generateGuid } from '../../utils';
+import { IAzureResourceService } from '../../interfaces';
+import { ResourceTreeDataProviderBase } from '../resourceTreeDataProviderBase';
+import { azureResource } from '../../azure-resource';
+
+export class PostgresServerArcTreeDataProvider extends ResourceTreeDataProviderBase<azureResource.AzureResourceDatabaseServer> {
+	private static readonly containerId = 'azure.resource.providers.postgresArcServer.treeDataProvider.postgresServerContainer';
+	private static readonly containerLabel = localize('azure.resource.providers.postgresArcServer.treeDataProvider.postgresServerContainerLabel', "PostgreSQL Hyperscale - Azure Arc");
+
+	public constructor(
+		databaseServerService: IAzureResourceService<azureResource.AzureResourceDatabaseServer>,
+		apiWrapper: ApiWrapper,
+		private _extensionContext: ExtensionContext
+	) {
+		super(databaseServerService, apiWrapper);
+	}
+
+
+	protected getTreeItemForResource(databaseServer: azureResource.AzureResourceDatabaseServer): TreeItem {
+		return {
+			id: `databaseServer_${databaseServer.id ? databaseServer.id : databaseServer.name}`,
+			label: databaseServer.name,
+			// TODO: should get PGSQL-specific icons (also needed in that extension)
```

Comment:
please create an issue for this if it doesn't exist already

Reply:
https://github.com/microsoft/azuredatastudio/issues/5269